app/src/lib/editor/engine/index.ts

```diff
@@ -44,20 +46,49 @@ export class EditorEngine {
         this.editorMode = mode;
     }
 
-    updateStyle(style: string, value: string) {
-        this.state.selected.forEach((elementMetadata) => {
+    private updateStyle(targets: UpdateStyleAction['targets'], style: string, value: string) {
+        targets.forEach((elementMetadata) => {
             const webview = this.webviews.get(elementMetadata.webviewId);
             if (!webview) {
                 return;
             }
             webview.send(WebviewChannels.UPDATE_STYLE, {
                 selector: elementMetadata.selector,
-                style,
-                value,
+                style: style,
+                value: value,
             });
         });
     }
 
+    private dispatchAction(action: Action) {
```

Comment:
Refactoring notes:
Thoughts on exposing the history object such that the EditorEngine API is cleaner?
`editor.history.runAction vs editor.runAction`

Or alternatively, having an action object that manages actions, transaction state, and history.  `editor.actions.run(action)`. 

Very minor, 2-way door decisions.

Reply:
yeah that sounds good, added it to the todo list

Reply:
Pushed it in https://github.com/onlook-dev/studio/pull/54/commits/70f843498e975aa4023737057329a378f6554ba8. I didn't put `runAction` on the new property because it's not history-specific, but I put `*Transaction` and `undo` in there